### PR TITLE
[HWKMETRICS-190] Fetch tenantIds from the metrics_idx table also

### PR DESCRIPTION
**Pull request by burmanm**
Fetches the tenant_ids from metrics_idx table as well as tenant table and then returns the tenants.

### CODE REVIEW

File: core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/MetricsServiceImpl.java [hunk 1]
```diff
@@ -395,15 +395,20 @@ public void setDateTimeService(DateTimeService dateTimeService) {
         return dataAccess.findAllTenantIds()
                 .flatMap(Observable::from)
                 .map(row -> row.getString(0))
-                .flatMap(dataAccess::findTenant)
-                .flatMap(Observable::from)
-                .map(Functions::getTenant);
+                .distinct()
```
Comment (jsanda) [inline core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/MetricsServiceImpl.java]: Is this ```distinct()``` call doing anything useful? Isn't is operating on a com.datastax.driver.core.Row object? I believe com.datastax.driver.core.ArrayBackedRow is the only Row implementation, and it does not implement ```equals`` in which case every row object would be considered distinct, right?

**Comment by stefannegrea** [comment]
Please do not merge this PR until 0.6.0 has been released.

File: core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/MetricsServiceImpl.java (hunk 1 above)
Comment (burmanm) [inline core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/MetricsServiceImpl.java]: No, it's calling equals on the String methods (couple lines above we do map(row -> row.getString(0)), which is the tenantId. There's a test in the MetricsServiceImpl that checks that no duplicate lines are returned.

**Comment by jsanda** [comment]
The changes look good to me but checkstyle is not happy due to unused import in MetricsServiceImpl.

**Comment by burmanm** [comment]
However, the unused import is actually used?

"/home/travis/build/hawkular/hawkular-metrics/core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/MetricsServiceImpl.java:24:15: Unused import - org.hawkular.metrics.core.api.MetricType.COUNTER_RATE."

But line 654 in MetricsServiceImpl (findRateData):

MetricId rateId = new MetricId(id.getTenantId(), COUNTER_RATE, id.getName(), id.getInterval());

So it's there. Buggy checkstyle (this error does not happen on my machine when compiling) or buggy travis compiling wrong PR?

**Comment by jsanda** [comment]
I am going to merge this even though the perf tests are failing. I have spent time discussing the perf tests with Filip and determined the causes. I will address it in a separate ticket/PR.

**Pull request merged**